### PR TITLE
Integration editing of geometries with new featuregrid

### PR DESCRIPTION
**Pull request by MV88**

### CODE REVIEW

File: web/client/components/map/leaflet/DrawSupport.jsx [hunk 1]
```diff
@@ -98,11 +114,23 @@ class DrawSupport extends React.Component {
             radius: radius,
             projection: projection
         };
-
-        this.props.onChangeDrawingStatus('stop', this.props.drawMethod, this.props.drawOwner);
+        /*if (this.props.options.editEnabled) {
```
Comment (offtherailz) [inline web/client/components/map/leaflet/DrawSupport.jsx]: remove

File: web/client/components/map/leaflet/DrawSupport.jsx [hunk 2]
```diff
@@ -125,40 +153,51 @@ class DrawSupport extends React.Component {
...
+            this.addGeojsonLayer({features: newProps.features, projection: newProps.options && newProps.options.featureProjection || "EPSG:4326"});
         } else {
             this.drawLayer.clearLayers();
             this.drawLayer.addData(this.convertFeaturesPolygonToPoint(newProps.features, this.props.drawMethod));
         }
     };
 
     addDrawInteraction = (newProps) => {
-        if (!this.drawLayer) {
+        /*if (!this.drawLayer) {
```
Comment (offtherailz) [inline web/client/components/map/leaflet/DrawSupport.jsx]: remove, we use git for versioning

File: web/client/components/map/leaflet/DrawSupport.jsx [hunk 3]
```diff
@@ -203,24 +242,79 @@ class DrawSupport extends React.Component {
...
         }
 
         // start the draw control
         this.drawControl.enable();
     };
 
+    addEditInteraction = (newProps) => {
+        this.addGeojsonLayer({features: newProps.features, projection: newProps.options && newProps.options.featureProjection || "EPSG:4326"});
+
+        /*if (newProps.options.drawEnabled) {
```
Comment (offtherailz) [inline web/client/components/map/leaflet/DrawSupport.jsx]: remove

File: web/client/components/map/leaflet/DrawSupport.jsx [hunk 4]
```diff
@@ -240,6 +334,26 @@ class DrawSupport extends React.Component {
         }) : features;
 
     };
+    fromLeafletFeature = (feature) => {
```
Comment (offtherailz) [inline web/client/components/map/leaflet/DrawSupport.jsx]: seems not used, please check

File: web/client/plugins/Map.jsx [hunk 5]
```diff
@@ -19,7 +19,7 @@ const Message = require('../components/I18N/Message');
 const ConfigUtils = require('../utils/ConfigUtils');
 const {isString} = require('lodash');
 let plugins;
-const {handleCreationLayerError, handleCreationBackgroundError} = require('../epics/map');
+const {handleCreationLayerError, handleCreationBackgroundError, startEditing} = require('../epics/map');
```
Comment (offtherailz) [inline web/client/plugins/Map.jsx]: startEditingGeometry

File: web/client/reducers/featuregrid.js [hunk 6]
```diff
@@ -56,6 +57,8 @@ function featuregrid(state = emptyResultsState, action) {
         return assign({}, state, {
             select: state.select.filter(f1 => !isPresent(f1, action.features))
             });
+    case DELETE_SELECTED_FEATURES_CONFIRM: // TODO implement this reducer
```
Comment (offtherailz) [inline web/client/reducers/featuregrid.js]: remove

File: web/client/reducers/featuregrid.js [hunk 7]
```diff
@@ -17,6 +17,7 @@ const {
     SAVE_SUCCESS,
     SAVE_ERROR,
     CLEAR_CHANGES,
+    DELETE_SELECTED_FEATURES_CONFIRM,
```
Comment (offtherailz) [inline web/client/reducers/featuregrid.js]: remove

File: web/client/epics/map.js [hunk 8]
```diff
@@ -61,7 +66,45 @@ const handleCreationLayerError = (action$, store) =>
         ]) : Rx.Observable.empty();
     });
 
+const startEditing = (action$, store) =>
+    action$.ofType(FEATURE_EDITING)
+    .filter(() => store.getState().featuregrid && store.getState().featuregrid.select && store.getState().featuregrid.select.length > 0)
```
Comment (offtherailz) [inline web/client/epics/map.js]: not needed

File: web/client/epics/map.js [hunk 9]
```diff
@@ -61,7 +66,45 @@ const handleCreationLayerError = (action$, store) =>
...
+    .switchMap(() => {
+        const isLeaflet = store.getState().maptype.mapType === "leaflet";
+        const defaultFeatureProj = "EPSG:4326";
+        let newFeatures;
+        let feature = head(store.getState().featuregrid.select); // just to be sure there is one feature selected for the editing
+        let drawOptions = {
+            featureProjection: defaultFeatureProj,
+            stopAfterDrawing: true,
```
Comment (offtherailz) [inline web/client/epics/map.js]: these should be deduced from the action payload (e.g. `create` or `edit`  flags)

File: web/client/epics/map.js [hunk 10]
```diff
@@ -61,7 +66,45 @@ const handleCreationLayerError = (action$, store) =>
...
     });
 
+const startEditing = (action$, store) =>
+    action$.ofType(FEATURE_EDITING)
+    .filter(() => store.getState().featuregrid && store.getState().featuregrid.select && store.getState().featuregrid.select.length > 0)
+    .switchMap(() => {
+        const isLeaflet = store.getState().maptype.mapType === "leaflet";
+        const defaultFeatureProj = "EPSG:4326";
+        let newFeatures;
+        let feature = head(store.getState().featuregrid.select); // just to be sure there is one feature selected for the editing
```
Comment (offtherailz) [inline web/client/epics/map.js]: get it from the action payload

File: web/client/epics/map.js [hunk 11]
```diff
@@ -61,7 +66,45 @@ const handleCreationLayerError = (action$, store) =>
         ]) : Rx.Observable.empty();
     });
 
+const startEditing = (action$, store) =>
+    action$.ofType(FEATURE_EDITING)
```
Comment (offtherailz) [inline web/client/epics/map.js]: use an intermediation between the "geometry editing service" provided by the draw support and the editing that the featuregrid triggers.

**Comment by MV88** [comment]
closing this and opening a new one

**Pull request closed without merging**